packages/react-bindings/src/telemetry/useTelemetry.ts

```diff
@@ -0,0 +1,40 @@
+import { Telemetry } from './types'
+
+const useTelemetry = (displayName: string, telemetry: Telemetry | undefined) => {
+  let start: number = -1
+  let end: number = -1
+
+  const setStart = () => {
+    start = telemetry && telemetry.enabled ? performance.now() : -1
+  }
+
+  const setEnd = () => {
+    if (telemetry && telemetry.enabled && start !== -1) {
+      end = performance.now()
+      const duration = end - start
+      if (telemetry.performance[displayName]) {
+        telemetry.performance[displayName].count++
+        telemetry.performance[displayName].msTotal += duration
+        telemetry.performance[displayName].msMin = Math.min(
+          duration,
+          telemetry.performance[displayName].msMin,
+        )
+        telemetry.performance[displayName].msMax = Math.max(
+          duration,
+          telemetry.performance[displayName].msMax,
+        )
+      } else {
+        telemetry.performance[displayName] = {
+          count: 1,
+          msTotal: duration,
+          msMin: duration,
+          msMax: duration,
+        }
+      }
+    }
+  }
+
+  return [setStart, setEnd]
```

Comment:
I think we should consider a pattern for hooks return values:

If you are returning a value and a setter, use an array to return a double. The problem with arrays is that they convey nothing specific about the return values - no names, no types, no quantity. Just things.

If you are returning arbitrary things, use an object. You can return clear names that make sense. Quantity of things returned is clear.

```tsx
return { setStart, setEnd }
```

Reply:
Agreed, will update it. @layershifter let's to this with the other hooks too.

Reply:
The hook was moved to a separate PR, and all PRs which don't follow the [value, setter] rule are returning objects: https://github.com/microsoft/fluent-ui-react/pull/2257